lib/pbench/test/unit/agent/test_tool_meister.py

```diff
@@ -22,92 +56,98 @@ def get_params():
-
-def test_create_tar_ignore_warnings(agent_setup, get_params):
-    """Test if we can suppress the errors raised during the tar creation"""
-    logger = get_logger("__logger__")
-    tm = ToolMeister(
-        pbench_install_dir=None,
-        tmp_dir=None,
-        tar_path="/usr/bin/tar",
-        sysinfo_dump=None,
-        params=get_params,
-        redis_server=None,
-        logger=logger,
-    )
-    tmp_dir = agent_setup["tmp"]
-
-    # create a file in tmp directory
-    (tmp_dir / "file1").write_text("")
-
-    target_dir = tmp_dir.name
-    tar_file = tmp_dir / f"{target_dir}.tar.xz"
-
-    cp = tm._create_tar(tmp_dir, tar_file)
-    assert cp.returncode == 1
-    assert b"file changed as we read it" in cp.stdout
-
-    cp = tm._create_tar(tmp_dir, tar_file, retry=True)
-    assert cp.returncode == 0
-    assert cp.stdout == b""
-    tar_file.unlink()
-
-
-def test_create_empty_tar(agent_setup, get_params):
-    """Test empty tar creation"""
-    logger = get_logger("__logger__")
-    tm = ToolMeister(
-        pbench_install_dir=None,
-        tmp_dir=None,
-        tar_path="/usr/bin/tar",
-        sysinfo_dump=None,
-        params=get_params,
-        redis_server=None,
-        logger=logger,
-    )
-    tmp_dir = agent_setup["tmp"]
-
-    # create a file in tmp directory
-    (tmp_dir / "file1").write_text("")
-
-    target_dir = tmp_dir.name
-    tar_file = tmp_dir / f"{target_dir}.tar.xz"
-
-    cp = tm._create_tar(pathlib.Path("/dev/null"), tar_file)
-    assert cp.returncode == 0
-    assert cp.stdout == b""
-
-    tar = tarfile.open(tar_file)
-    # There should be only one member inside this tar file i.e. null
-    for member in tar.getmembers():
-        assert member.name == "null"
-
-    tar_file.unlink()
+
+    def test_create_tar(self, agent_setup, mock_tar):
+        """Test create tar file"""
+        tm = ToolMeister(
+            pbench_install_dir=None,
+            tmp_dir=None,
+            tar_path="tar_path",
+            sysinfo_dump=None,
+            params=TestCreateTar.tm_params,
+            redis_server=None,
+            logger=None,
+        )
+        tmp_dir = agent_setup["tmp"]
+        tar_file = "test.tar.xz"
+
+        cp = tm._create_tar(tmp_dir, pathlib.Path(tar_file))
+        assert cp.returncode == 0
+        assert cp.stdout == b""
+
+    def test_create_tar_ignore_warnings(self, agent_setup, mock_tar_no_warnings):
+        """Test if we can suppress the errors raised during the tar creation"""
+        tm = ToolMeister(
+            pbench_install_dir=None,
+            tmp_dir=None,
+            tar_path="tar_path",
+            sysinfo_dump=None,
+            params=TestCreateTar.tm_params,
+            redis_server=None,
+            logger=logger,
+        )
+        tmp_dir = agent_setup["tmp"]
+        tar_file = "test.tar.xz"
+
+        cp = tm._create_tar(tmp_dir, pathlib.Path(tar_file))
+        assert cp.returncode == 1
+        assert cp.stdout == b"Some error running tar"
```

Comment:
It would be better if there were a common definition for what the mock returns and for what we test against here (i.e., at the module level, define a symbol for the error message string, and then reference that symbol in the mocks and here).  Also, it would be good if the messages from the first try and the retry were different (if you make them so that they can both fail), so that we can ensure that we're seeing the correct one.